docs/code/VBufferCareFeeding.md

```diff
@@ -83,57 +72,62 @@ not commutative, operations over sparse `VBuffer<float>` or `VBuffer<double>`
 vectors can sometimes result in modestly different results than the "same"
 operation over dense values.
 
-## Why Buffer Reuse
+There is also another point about sparsity, that is, about the increasing
+indices. While it is a *requirement*, it is regrettably not one we can
+practically enforce without a heavy performance cost. So, if you are creating
+your own `VBuffer`s, be careful that you are creating indices that are in
+strictly increasing order, and in the right range (non-negative and less than
+`Length`). We cannot check!
+
+# Buffer Reuse
 
 The question is often asked by people new to this codebase: why bother with
 buffer reuse at all? Without going into too many details, we used to not and
-suffered for it. We had a far simpler system where examples were yielded
-through an
-[`IEnumerable<>`](https://msdn.microsoft.com/en-us/library/9eekhta0.aspx), and
-our vector type at the time had `Indices` and `Values` arrays as well, but
-their sizes were there actual sizes, and being returned through an
-`IEnumerable<>` there was no plausible way to "recycle" the buffers.
-
-Also: who "owned" a fetched example (the caller, or callee) was not clear.
-Because it was not clear, code was inevitably written and checked in that made
+suffered for it. Long ago we had a far simpler system where examples were
+yielded through an [`IEnumerable<>`][1], and our vector type at the time had
+`Indices` and `Values`, but exposed as arrays, and their sizes were the actual
+"logical" sizes, and being returned through an `IEnumerable<>` there was no
+plausible way to reuse buffers.
+
+Also: who "owned" a fetched example (the caller, or callee) was not clear;
+code was at one time written under both assumptions, which was a mess. Because
+it was not clear, code was inevitably written and checked in that made
 *either* assumption, which meant, ultimately, that everything that touched
-these would try to duplicate everything by default, because doing anything
-else would fail in some case.
-
-The reason why this becomes important is because [garbage
-collection](https://msdn.microsoft.com/en-us/library/0xy59wtx.aspx) in the
-.NET framework is not free. Creating and destroying these arrays *can* be
-cheap, provided that they are sufficiently small, short lived, and only ever
-exist in a single thread. But, violate any of these, there is a possibility
-these arrays could be allocated on the large object heap, or promoted to gen-2
+these would try to duplicate everything, because doing anything else would
+fail in *some* case.
+
+The reason why this becomes important is because [garbage collection][2] in
+.NET is not free. Creating and destroying these arrays *can* be cheap,
+provided that they are sufficiently small, short lived, and only ever exist in
+a single thread. But, if that ever does not hold, there is a possibility these
```

Comment:
`But, if that ever does not hold` 

maybe could be rewritten as

`But, when that does not hold`

Reply:
That is better -- "if that ever" suggests the occurrence is rare, whereas "when" doesn't have the same implicit suggestion, thanks @eerhardt.